sdk/samples/iot/pnp/pnp_thermostat_component.c

```diff
@@ -50,66 +50,80 @@ static char command_end_time_value_buffer[32];
 // IoT Hub Telemetry Values
 static az_span const telemetry_temperature_name = AZ_SPAN_LITERAL_FROM_STR("temperature");
 
-static az_result build_command_response_payload(
```

Comment:
This file includes az_result.h. It looks like it is no longer needed. Please check other sample files, if it is the case with them too. Probably, only the file that defines the `IOT_SAMPLE_EXIT_IF_AZ_FAILED` needs to include it; also any file that uses az_result itself, any specific value, such as AZ_OK, or the az_result_failed()/succeeded(), if there are any left.